### PR TITLE
SPMD communicator profiler

### PR DESCRIPTION
**Pull request by VadimVoevodkin**

### CODE REVIEW

File: cpp/oneapi/dal/test/ccl_communicator.cpp [hunk 1]
```diff
@@ -282,15 +282,15 @@ TEST_M(ccl_comm_test, "allgatherv_empty_rank") {
...
-#endif
+// #ifdef ONEDAL_DATA_PARALLEL
+//     SECTION("device") {
+//         test_allgatherv_on_device(send_buffer.data(),
+//                                   rank_size,
+//                                   recv_buffer.data(),
+//                                   recv_counts.data(),
+//                                   displs.data());
+//     }
+// #endif
```
Comment (VadimVoevodkin) [inline cpp/oneapi/dal/test/ccl_communicator.cpp:293]: commented out until the oneCCL bug is fixed

**Comment by VadimVoevodkin** [comment]
intelci: run

**Comment by VadimVoevodkin** [comment]
intelci: run

File: cpp/oneapi/dal/detail/ccl/communicator.hpp [hunk 2]
```diff
@@ -185,14 +187,22 @@ class ccl_device_communicator_impl : public spmd::communicator_iface {
             internal_recv_counts[i] = integral_cast<size_t>(recv_counts[i]);
         }
 
+        const std::int64_t data_type_size = get_data_type_size(dtype);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/detail/ccl/communicator.hpp:190]: Please don't mix profiling and code changes

File: cpp/oneapi/dal/detail/ccl/communicator.hpp [hunk 3]
```diff
@@ -175,6 +176,7 @@ class ccl_device_communicator_impl : public spmd::communicator_iface {
         preview::detail::check_if_pointer_matches_queue(queue_, send_buf);
         preview::detail::check_if_pointer_matches_queue(queue_, recv_buf);
 
+        ONEDAL_PROFILER_TASK(CCL_allgatherv, q);
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/detail/ccl/communicator.hpp:179]: Not in accordance with last decision

File: cpp/oneapi/dal/test/ccl_communicator.cpp [hunk 4]
```diff
@@ -282,15 +282,15 @@ TEST_M(ccl_comm_test, "allgatherv_empty_rank") {
...
-#ifdef ONEDAL_DATA_PARALLEL
-    SECTION("device") {
-        test_allgatherv_on_device(send_buffer.data(),
-                                  rank_size,
-                                  recv_buffer.data(),
-                                  recv_counts.data(),
-                                  displs.data());
-    }
-#endif
+    // #ifdef ONEDAL_DATA_PARALLEL
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/test/ccl_communicator.cpp:285]: Please don't mix new tests and profiling

**Comment by KulikovNikita** [comment]
It is done in a different way now.

**Pull request closed without merging**